src/types.h

```diff
@@ -478,8 +478,13 @@ class SizedType {
   }
   bool IsMapIterableTy() const
   {
-    return !(type_ == Type::hist_t || type_ == Type::lhist_t ||
-             type_ == Type::stats_t);
+    return !IsRestrictedMapTy();
+  }
+
+  bool IsRestrictedMapTy() const
```

Comment:
"restricted" is a bit ambiguous. what restriction? can the existing `IsPrintableTy()` be extended for this?

Reply:
Yeah I couldn't figure out a better name. Unfortunately we can't use `IsPrintableTy` because this restriction also applies to adding these types to tuples.

Reply:
Maybe I can utilize the new semantic_analyser function `IsValidAssignment` which checks these same types.

Reply:
Can't you call IsPrintableTy() form inside the Tuple check?

Reply:
Ok, changed the name to "IsMultiOutputMapTy" - WDYT?